internal/support/scheduler/utils.go

```diff
@@ -14,25 +14,26 @@
 package scheduler
 
 import (
-	"regexp"
 	"strconv"
 	"time"
 )
 
-const (
-	frequencyPattern = `^P(\d+Y)?(\d+M)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$`
-)
-
+// Frequency indicates how often the specific resource needs to be polled.
+// It represents as a duration string.
+// The format of this field is to be an unsigned integer followed by a unit which may be "ms", "s", "m" or "h"
+// representing milliseconds, seconds, minutes or hours. Eg, "100ms", "24h"
 func isFrequencyValid(frequency string) bool {
```

Comment:
This function is simple enough that I think we should refactor it out. It's not adding a whole lot of value.

Reply:
I left it because it's being used by other potential functions.

Reply:
Then remove it there too.

Reply:
after our discussion I will be refactoring this util function out and just modify the code to use the parseDuration func :)